src/offset/local/unix.rs

```diff
@@ -37,35 +37,89 @@ pub(super) fn naive_to_local(d: &NaiveDateTime, local: bool) -> LocalResult<Date
 }
 
 fn offset(d: NaiveDateTime, local: bool) -> LocalResult<FixedOffset> {
-    let info = unsafe {
-        INIT.call_once(|| {
-            INFO = Some(TimeZone::local().expect("unable to parse localtime info"));
-        });
-        INFO.as_ref().unwrap()
-    };
+    TZINFO.with(|cache| {
```

Comment:
Let's move all this code into a `&self` method on `Cache`?

Reply:
good call. Hopefully I've struck the right balance of what I moved and what I've left in `naive_to_local`